### PR TITLE
refactor(consolidation): replace usize::MAX dedup sentinel with DedupOutcome enum

### LINKED ISSUE #142: [super-qa] medium: usize::MAX sentinel leaks via public local_dedup API
## Finding: Magic Sentinel in Public API

- **Severity:** medium
- **Category:** design
- **Location:** src/consolidation/dedup.rs:43
- **Detected by:** Codex review (super-review round 2)

## Description

\`local_dedup()\` returns \`usize::MAX\` as a "skipped" sentinel, but it's a \`pub\` function whose docs promise the first tuple element is the number of duplicates removed. External callers will see an undocumented magic value.

## Suggested Approach

Either:
1. Make \`consolidation\` module \`pub(crate)\` (tracked in #110)
2. Introduce an explicit return type: \`enum DedupResult { Completed { removed: usize, expired_ids: Vec<i64> }, Skipped }\`

## Related Issues

- #110 (all modules are pub)
- #115 (stringly-typed errors)

### LINKED ISSUE #272: [super-qa] high: Magic usize::MAX sentinel encodes 'skipped' status in numeric return value
> ⚠️ Generated by `/super-qa` 2026-06-01. Severity & details from agent analysis; **not independently source-verified**. Review before action.

**Magic usize::MAX sentinel encodes 'skipped' status in numeric return value**

- **ID:** `consolidation/design-usize-max-sentinel`
- **Severity:** high
- **Area:** consolidation · **Type:** refactor
- **Location:** `src/consolidation/dedup.rs:43-45, src/consolidation/mod.rs:55-57`
- **Auto-fixable:** no

## Description
Current state: `local_dedup` returns `Ok((usize::MAX, vec![]))` when the corpus exceeds `MAX_DEDUP_FACTS`, using a sentinel numeric value to communicate 'skipped' status to the caller. The orchestrator (`mod.rs:55-57`) then checks `duplicates_removed == usize::MAX`, decodes the sentinel back to 0, and introduces a parallel `dedup_skipped: bool` to gate the watermark advance. This is a stringly-typed/magic-value anti-pattern: the semantics are invisible in the type, the sentinel is incompatible with any future count ≥ `usize::MAX - 1`, and the decoding logic is split across two files. Target state: replace `(usize, Vec<i64>)` with an enum `enum DedupOutcome { Ran { removed: usize, expired_ids: Vec<i64> }, Skipped { active_count: usize } }`. The orchestrator pattern-matches the result; the `Skipped` arm carries diagnostic context (`active_count`) and the watermark logic becomes an explicit match arm with no decoding arithmetic. Refactoring pattern: Replace Magic Number with Type. Motivation: correctness (magic value collides with legitimate count), readability (intent hidden from call site), testability (can assert `Skipped` variant without knowing the sentinel). Scope: 2 files (`dedup.rs`, `mod.rs`). Prerequisites: full test coverage of the dedup-skipped path (existing tests do not cover it). Risk: medium — touches the return signature of a `pub(crate)` function.

## Suggested approach
// dedup.rs — replace the (usize, Vec<i64>) return with:
pub enum DedupOutcome {
    Ran { removed: usize, expired_ids: Vec<i64> },
    Skipped { active_count: usize },
}
// mod.rs — replace the sentinel check:
let (duplicates_removed, expired_ids, dedup_skipped) = match local_dedup(...)? {
    DedupOutcome::Ran { removed, expired_ids } => (removed, expired_ids, false),
    DedupOutcome::Skipped { .. } => (0, vec![], true),
};

## References
—

### LINKED ISSUE #306: [super-qa] high: consolidate() orchestrator has no #[cfg(test)] module
> ⚠️ Generated by `/super-qa` 2026-06-01. Severity & details from agent analysis; **not independently source-verified**. Review before action.

**consolidate() orchestrator has no #[cfg(test)] module**

- **ID:** `consolidation/testing-no-inline-tests-orchestrator`
- **Severity:** high
- **Area:** consolidation · **Type:** test
- **Location:** `src/consolidation/mod.rs:35-78`
- **Auto-fixable:** no

## Description
The top-level `consolidate()` function (mod.rs:35) is responsible for: reading/writing the `last_consolidated_at` watermark, detecting the `usize::MAX` sentinel that suppresses watermark advancement when dedup is skipped, and committing a single atomic transaction across all three passes. None of these behaviors have unit tests. The sentinel detection (mod.rs:55-66) and the conditional `set_config` call (mod.rs:64-66) are completely untested at the unit level. Integration tests in `tests/eval/quality/consolidation.rs` exercise the happy path but do not test watermark suppression.

## Suggested approach
Add a `#[cfg(test)] mod tests` block to mod.rs. At minimum: (1) a test that calls `consolidate()` on an empty DB and asserts `last_consolidated_at` is set; (2) a test that mocks or simulates the over-cap condition (by inserting via a stub or directly calling `local_dedup` with the sentinel return) and asserts `last_consolidated_at` is NOT advanced; (3) a test that uses a failing `SummaryGenerator` and asserts the DB is unchanged (transaction rolled back) after the call returns `Err`.

## References
—

### LINKED ISSUE #345: [super-qa] medium: Safety cap constant duplicated in dedup and cluster with inconsistent enforcement policy
> ⚠️ Generated by `/super-qa` 2026-06-01. Severity & details from agent analysis; **not independently source-verified**. Review before action.

**Safety cap constant duplicated in dedup and cluster with inconsistent enforcement policy**

- **ID:** `consolidation/maintainability-duplicate-safety-cap-constants`
- **Severity:** medium
- **Area:** consolidation · **Type:** refactor
- **Location:** `src/consolidation/dedup.rs:30, src/consolidation/cluster.rs:36`
- **Auto-fixable:** no

## Description
Current state: `MAX_DEDUP_FACTS: usize = 50_000` (dedup.rs:30) and `MAX_CLUSTER_FACTS: usize = 50_000` (cluster.rs:36) are identical values in two files. More importantly, they are enforced differently: the dedup cap returns a sentinel and blocks watermark advancement (skip-with-consequence); the cluster cap silently returns 0 (skip-with-no-consequences, existing summaries are deleted at line 53 after the cap check). A future developer changing one cap is unlikely to know about the other, and the two enforcement policies diverge silently. Target state: lift both constants to `consolidation/mod.rs` or a private `constants` submodule; name them `MAX_FACTS_FOR_PAIRWISE_OPS` or with distinct, clear names; document the policy difference explicitly. Refactoring pattern: Consolidate Duplicate Constants / Introduce Explaining Variable. Motivation: maintainability (single source of truth), readability (explicit policy difference). Scope: 2 files. Prerequisites: none. Risk: low.

## Suggested approach
// consolidation/mod.rs (or a private constants.rs)
/// Safety cap for O(N·M) dedup. Beyond this, dedup is skipped
/// and the watermark is NOT advanced (retried next run).
pub(super) const MAX_FACTS_FOR_DEDUP: usize = 50_000;
/// Safety cap for O(N²) clustering. Beyond this, clustering is skipped
/// (existing summaries are preserved, not deleted).
pub(super) const MAX_FACTS_FOR_CLUSTERING: usize = 50_000;
// Reference these in dedup.rs and cluster.rs via super::

## References
—

### LINKED ISSUE #439: [super-qa] medium: Watermark suppression when dedup is skipped (usize::MAX sentinel) has no integration test
> ⚠️ Generated by `/super-qa` 2026-06-01. Severity & details from agent analysis; **not independently source-verified**. Review before action.

**Watermark suppression when dedup is skipped (usize::MAX sentinel) has no integration test**

- **ID:** `consolidation/testing-no-watermark-suppression-integration-test`
- **Severity:** medium
- **Area:** consolidation · **Type:** test
- **Location:** `src/consolidation/mod.rs:55-66`
- **Auto-fixable:** no

## Description
When `local_dedup` returns `usize::MAX` (over-cap path, dedup.rs:45), `consolidate()` must NOT advance `last_consolidated_at` so skipped facts are retried later. This logic (mod.rs:55-66) is critical for correctness: if the watermark were advanced despite the skip, those facts would be silently excluded from the next dedup window. No test exercises this path end-to-end via `engine.consolidate()`. The sentinel contract is documented only in a comment.

## Suggested approach
Add an integration test that: (1) inserts at least two near-duplicate facts, (2) directly manipulates the store to make `list_active` return a list longer than `MAX_DEDUP_FACTS` (e.g., by temporarily lowering the cap via a feature flag or by inserting inline via `FactStore`), (3) calls `consolidate()`, and (4) asserts `get_config("last_consolidated_at")` is still `None` (or unchanged from before). Because 50_000 is too large to insert in a test, a better approach is to expose the cap as a testable constant or add a `#[cfg(test)]` override path.

## References
—

### PR DESCRIPTION
**Pull request by dutiona**
## What

Wave 2 of the **#253 consolidation super-qa sweep**. Replaces the `usize::MAX` "skipped" sentinel in the dedup pass with an explicit `DedupOutcome` sum type, lifts + de-duplicates the safety caps, and adds the missing skip-path test coverage.

## Changes

- **#272** — `local_dedup` returns `enum DedupOutcome { Ran { removed, expired_ids }, Skipped { active_count } }` instead of `(usize, Vec<i64>)` with a `usize::MAX` magic first element. The orchestrator pattern-matches; the `Skipped` arm carries the tripping `active_count`; watermark-suppression is now an explicit match arm, not `== usize::MAX` decode arithmetic. The magic value could collide with any legitimate count `>= usize::MAX - 1`.
- **#142** — the sentinel no longer exists on the `local_dedup` API surface (the type change resolves it; consolidation is already `pub(crate)` so there was never external exposure, only the in-band magic value).
- **#345** — the two duplicated `MAX_*_FACTS = 50_000` caps are lifted to named constants (`MAX_FACTS_FOR_DEDUP` / `MAX_FACTS_FOR_CLUSTERING`) in `mod.rs` with their **divergent skip policies documented side by side** (dedup skip blocks the watermark → retried; cluster skip is a no-op preserving prior summaries).
- The dedup cap is now an injected parameter; `consolidate_with_caps` is the cap-injecting core, while the public `consolidate` signature is **unchanged** (engine/cli/mcp callers untouched). This makes the skip path testable without a 50 000-fact corpus.
- **#439 / #306** — `over_cap_returns_skipped_without_expiring` (unit: asserts `Skipped`) + `watermark_not_advanced_when_dedup_skipped` (orchestrator: watermark held on skip). #306's other two cases (empty-DB advance, failing-pass rollback) were already covered.

## Verification

- Behavior-preserving on the `Ran` path: full prior consolidation suite passes unchanged.
- `cargo test --workspace --all-features`: 882 root + all crate suites pass. `clippy --workspace --all-targets --all-features` + `fmt --check` clean.

Closes #272. Closes #142. Closes #345. Closes #439. Closes #306. Part of #253.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/consolidation/dedup.rs [hunk 1]
```diff
@@ -29,27 +53,24 @@ pub fn local_dedup(
...
             "dedup skipped: too many active facts for O(N*M) comparison; \
              watermark will NOT advance so skipped facts are retried when corpus shrinks"
         );
-        // Return sentinel value usize::MAX to signal "skipped" to the orchestrator.
-        // The orchestrator must NOT advance last_consolidated_at in this case.
-        return Ok((usize::MAX, vec![]));
+        return Ok(DedupOutcome::Skipped {
+            active_count: active_facts.len(),
+        });
     }
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/consolidation/dedup.rs:82]: Good catch — the over-cap path loads the full corpus before checking the cap. This is orthogonal to #272's sentinel→`DedupOutcome` refactor, so per the repo's collateral-issue policy I've filed it separately as #659 (linked under epic #253). It overlaps the Wave 5 (#389/#409) load restructuring and will be folded in there. Not blocking this PR.
